client/my-sites/woocommerce/dashboard/required-plugins-install-view.jsx

```diff
@@ -152,6 +153,41 @@ class RequiredPluginsInstallView extends Component {
 		}
 	};
 
+	createTimeoutTimer = () => {
+		if ( this.timeoutTimer ) {
+			return;
+		}
+
+		// Timeout setup after 25 seconds.
+		this.timeoutTimer = window.setTimeout( this.timeoutElapsed, 25000 );
+	};
+
+	timeoutElapsed = () => {
+		// These status means store setup is finished, not stalled.
+		if ( [ 'IDLE', 'DONESUCCESS', 'DONEFAILURE' ].includes( this.state.engineState ) ) {
+			return;
+		}
+
+		recordTrack( 'calypso_woocommerce_setup_timeout', {
```

Comment:
Instead of immediately failing, it would be good to retry things at least once. We have seen that refreshing the page often unblocks things, so it seems worthwhile to attempt another try to hopefully succeed and avoid a support interaction.

Reply:
@mattsherman I could not come up with a simple way to make a retry for each stage of the installation. The pattern used by `updateEngine` seems pretty odd and not react-like. A basic way is to reset the whole state and retry from step 1, and only attempt once of course. Otherwise, it looked very complicated in my mind. If agree or have any suggestion, please let me know!

Reply:
@ilyasfoo Hmm... I'd have to think about this more, but I believe that you are right. Let's not complicate matters right now -- don't worry about retrying in this PR. If we continue to hear if customer issues we can consider the best course of action at that point.

The whole way that state is handled in this code seems too complex/questionable for the client-side... ultimately I'd rather see all of this implemented in an API endpoint. But, that's for future consideration! 😄